packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1562,6 +1613,33 @@ class EditableText extends StatefulWidget {
   /// {@macro flutter.services.TextInputConfiguration.enableIMEPersonalizedLearning}
   final bool enableIMEPersonalizedLearning;
 
+  /// {@template flutter.widgets.editableText.contentCommitMimeTypes}
+  /// Used when a user inserts image-based content through the device keyboard,
+  /// currently only used on Android.
+  ///
+  /// The passed list of strings will determine which MIME types are allowed to
+  /// be inserted via the device keyboard. If no `onContentCommitted` callback
+  /// is provided, this field will be ignored.
+  ///
+  /// {@tool snippet}
+  /// The default mime types are given by [kDefaultContentCommitMimeTypes]. These are all the mime types
+  /// that are able to be handled and inserted from keyboards.
+  ///
+  /// {@end-tool}
+  ///
+  /// {@tool dartpad}
+  /// This example shows how to limit image insertion to specific file types.
+  ///
+  /// ** See code in examples/api/lib/widgets/editable_text/editable_text.on_content_committed.0.dart **
+  /// {@end-tool}
+  ///
+  /// See also:
+  ///
+  ///  * <https://developer.android.com/guide/topics/text/image-keyboard>
+  ///
+  /// {@endtemplate}
+  final List<String> contentCommitMimeTypes;
```

Comment:
Looks like `EditableText` isn't doing anything with the mime types, could you explain why this field is needed?

Reply:
This gets passed to the engine layer and set in `InputConnectionAdaptor`. This allows the user to disallow certain mime types, and the OS will actually pop a toast saying "the app doesn't support content insertion here".